cabal-install/Distribution/Client/RebuildMonad.hs

```diff
@@ -166,3 +176,58 @@ monitorDirectoryStatus dir = do
                     then monitorDirectory dir
                     else monitorNonExistentDirectory dir]
 
+doesFileExistMonitored :: FilePath -> Rebuild Bool
+doesFileExistMonitored f = do
+    root <- askRoot
+    exists <- liftIO $ doesFileExist (root </> f)
+    -- TODO: If the file exists, should we really monitor the entire
+    -- file?!
```

Comment:
Well whether we should `monitorFileHashed` or `monitorFile` depends on how the caller is going to want to use it. As it happens we always want one, but if we wanted different cases at different call sites then we'd want a param.